### ISSUE TITLE
dcs-tools use of sym links seems to mean it can't work on windows

### ISSUE THREAD

**Issue by nigelhaslam** [aktos-io/dcs-tools#14]
Hi,
I couldn't stop myself from going ahead and trying to get it working.

I set up my project as proposed in the advanced setup:

**home/
   remote-machines/
                             raspberrypiB/
                                        dcs-tools/**

But when I try to run the command
**./setup**

I get this error
**$ ./setup
ln: failed to create symbolic link '/c/Users/The Producer/remote-machines/raspberrypiB/dcs-tools/../Makefile': No such file or directory
There is a configuration file already**

I did find a file called config.sh, which looked as though it might be the intended result of the setup, so I added the IP address of my 'target', the RaspberryPi

However.. when I go on to run 
**$ ./dcs-tools/make-target-settings**

I get this error:
**bash: ./dcs-tools/make-target-settings: No such file or directory**

In researching the ./setup error I find that symbolic links may be the root of the problem:

**https://github.com/Microsoft/WSL/issues/353**

Does this seem likely to you?
Is there a workaround that you can suggest?
Other than me buying a linux workstation!

Thanks

**Comment by ceremcem** [comment]
Hi, 

Now this error makes me sense. Since you are on Windows, you are unable to use symlinks. This can be fixed, but there will be another requirements, such as: 

* A Linux file system, in order to let your backups has correct Unix file attributes (see https://github.com/aktos-io/dcs-tools#use-correct-filesystem)
* Since you are on Windows, you can not make benefit of hardlinks, where this tool uses for incremental backups. You simply can not have incremental backups. 
* Do you have `sshfs` which this tool uses under the hood when you `make mount-root`?
* Do you have `OpenSSH` which this tool uses when connecting to target via SSH?

Even if we make it work on Windows, you can only benefit from `make ssh` command. 

Your best bet might be setting up a virtual machine and using this tool in that environment.

**Comment by nigelhaslam** [comment]
Hi Cerem

Thanks for the heads up.

I wonder if there's a way to host a virtual machine on my PC?  I have a
powerful workstation..  i7 @4Ghz  32 Gig RAM etc.

The intention of the Raspberry pi is a remote time lapse camera control,
uploading images to the Internet via a 4G mobile connection.  Maybe a cloud
based linux  VPN server is the way to go?

I currently want to be uploading lower resolution jpeg images as peace of
mind and system security.. which would be light on the expensive data.
Would dcs-tools be able to exclude the higher resolution RAW folder, ir is
it an all or nothing sync?

Best wishes

N


On 26 Oct. 2017 6:54 pm, "Cerem Cem ASLAN" <notifications@github.com> wrote:

Hi,

Now this error makes me sense. Since you are on Windows, you are unable to
use symlinks. This can be fixed, but there will be another requirements,
such as:

   - A Linux file system, in order to let your backups has correct Unix
   file attributes (see https://github.com/aktos-io/dcs-tools#use-correct-
   filesystem)
   - Since you are on Windows, you can not make benefit of hardlinks, where
   this tool uses for incremental backups. You simply can not have incremental
   backups.
   - Do you have sshfs which this tool uses under the hood when you make
   mount-root?
   - Do you have OpenSSH which this tool uses when connecting to target via
   SSH?

Even if we make it work on Windows, you can only benefit from make ssh
command.

Your best bet might be setting up a virtual machine and using this tool in
that environment.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/aktos-io/dcs-tools/issues/14#issuecomment-339582730>,
or mute the thread

**Comment by ceremcem** [comment]
You can setup any virtual machine without any issue. My recommendation would be using Debian over Virtualbox though. 

You can exclude/include any folder and/or file types according to your needs. See https://github.com/aktos-io/dcs-tools/blob/master/exclude-list.txt

> The intention of the Raspberry pi is a remote time lapse camera control,
uploading images to the Internet via a 4G mobile connection.  Maybe a cloud
based linux  VPN server is the way to go?

This is not related with `dcs-tools` but you may really want to look at [`link-with-server`](https://github.com/aktos-io/link-with-server/) in this case, it will be the simplest possible solution I can think of.

**Comment by nigelhaslam** [comment]
Hi Cerem,
I looked at Virtualbox and then I looked for and couldn't find a cloud based option, which rather surprised me because it seems everything else is catered for with cloud based SAAS options.
I didn't move forward because I need a bit of time to regroup, and let the ideas settle in my head before I proceed.
I think the most useful and important thing to add to your usage README is that this is a linux only solution.
Cheers and have a great weekend

Nigel
